srv/src/main/java/my/bookshop/config/DestinationConfiguration.java

```diff
@@ -36,44 +36,37 @@ public class DestinationConfiguration {
 
 	@EventListener
 	void applicationReady(ApplicationReadyEvent ready) {
-		String applicationUrl = runtime.getEnvironment().getApplicationInfo().getUrl();
+		Optional<ServiceBinding> binding = runtime.getEnvironment().getServiceBindings()
+				.filter(b -> b.getServiceName().get().equals("xsuaa")).findFirst();
 
-		if (applicationUrl != null) {
-			// it seems we're running in the cloud
-			registerCloudDestination(applicationUrl);
-		} else {
-			registerLocalDestination();
-		}
+		binding.ifPresentOrElse(this::registerCloudDestination, this::registerLocalDestination);
 	}
 
-	private void registerCloudDestination(String applicationUrl) {
-		String destinationName = environment.getProperty("cds.remote.services.'[API_BUSINESS_PARTNER]'.destination.name");
+	private void registerCloudDestination(ServiceBinding xsuaaBinding) {
+		String destinationName = environment
+				.getProperty("cds.remote.services.'[API_BUSINESS_PARTNER]'.destination.name");
 
 		logger.info("Destination name for mocked API_BUSINESS_PARTNER: {}", destinationName);
+		logger.info("UAA Service Binding: {} / {}", xsuaaBinding.getName().get(), xsuaaBinding.getServiceName().get());
 
-		ServiceBinding uaaBinding = runtime.getEnvironment().getServiceBindings().filter(b -> b.getServiceName().get().equals("xsuaa")).findFirst().get();
-		logger.info("UAA Service Binding: {} / {}", uaaBinding.getName().get(), uaaBinding.getServiceName().get());
+		ServiceBindingDestinationOptions options = ServiceBindingDestinationOptions.forService(xsuaaBinding)
+				.withOption(BtpServiceOptions.AuthenticationServiceOptions.withTargetUri("http://localhost"))
```

Comment:
To be on the safe-side I think you should also add the port to the URL here, retrieved as for `registerLocalDestination()`. Not sure if this even works right now, as I think CF uses port 8080 by default and not 80.